File: Heat/ModelBuilders/CustomerModelViewBuilder.vb

Imports Heat.Models
Imports Heat.ViewModels.Customers
Imports AutoMapper

Public Class CustomerModelViewBuilder

    Private _db As IHeatDBContext

    Public Sub New(db As IHeatDBContext)
        _db = db
    End Sub

    Public Function GetIndexCustomerViewModel(includeDisabled As Boolean) As IndexCustomerViewModel
        Dim result As New IndexCustomerViewModel
        Dim customerList As List(Of Customer)

        If includeDisabled Then
            customerList = _db.Customers.OrderBy(Function(c) c.Name).ToList
            result.IsIncludeDisable = True
        Else
            customerList = _db.Customers.Where(Function(c) c.IsEnabled = True).OrderBy(Function(c) c.Name).ToList
            result.IsIncludeDisable = False
        End If

        result.HasDisabled = _db.Customers.Where(Function(c) c.IsEnabled = False).Count > 0
        result.Rows = Mapper.Map(Of List(Of IndexCustomerGridViewModel))(customerList)

        Return result

    End Function

    Public Function GetDisableCustomerViewModel(id As Integer) As DisableCustomerViewModel
        Dim result As New DisableCustomerViewModel
        Dim c As Customer

        c = _db.Customers.Find(id)
        result.ID = c.ID
        result.CustomerName = c.Name

        Return result

    End Function

    Public Function GetEnableCustomerViewModel(id As Integer) As EnableCustomerViewModel
        Dim result As New EnableCustomerViewModel
        Dim c As Customer
        c = _db.Customers.Find(id)

        result.ID = c.ID
        result.CustomerName = c.Name
        result.DisableDate = c.DisableDate

        Return result
    End Function
    Public Function GetEditCustomerViewModel(id As Integer) As EditCustomerViewModel
        Dim result As New EditCustomerViewModel
        Dim editingItem As Customer

        editingItem = _db.Customers.Find(id)

        result = Mapper.Map(Of EditCustomerViewModel)(editingItem)

        Return result
    End Function

End Class

File: Heat.ConvertedToC#/ModelBuilders/CustomerModelViewBuilder.cs
using System.Collections.Generic;
using System.Linq;

using Heat.Models;
using Heat.ViewModels.Customers;
using AutoMapper;
namespace Heat
{

    public class CustomerModelViewBuilder
	{


		private IHeatDBContext _db;
		public CustomerModelViewBuilder(IHeatDBContext db)
		{
			_db = db;
		}

		public IndexCustomerViewModel GetIndexCustomerViewModel(bool includeDisabled)
		{
			IndexCustomerViewModel result = new IndexCustomerViewModel();
			List<Customer> customerList = null;

			if (includeDisabled) {
				customerList = _db.Customers.OrderBy(c => c.Name).ToList();
				result.IsIncludeDisable = true;
			} else {
				customerList = _db.Customers.Where(c => c.IsEnabled == true).OrderBy(c => c.Name).ToList();
				result.IsIncludeDisable = false;
			}

			result.HasDisabled = _db.Customers.Where(c => c.IsEnabled == false).Count() > 0;
			result.Rows = Mapper.Map<List<IndexCustomerGridViewModel>>(customerList);

			return result;

		}

		public DisableCustomerViewModel GetDisableCustomerViewModel(int id)
		{
			DisableCustomerViewModel result = new DisableCustomerViewModel();
			Customer c = null;

			c = _db.Customers.Find(id);
			result.ID = c.ID;
			result.CustomerName = c.Name;

			return result;

		}

		public EnableCustomerViewModel GetEnableCustomerViewModel(int id)
		{
			EnableCustomerViewModel result = new EnableCustomerViewModel();
			Customer c = null;
			c = _db.Customers.Find(id);

			result.ID = c.ID;
			result.CustomerName = c.Name;
			result.DisableDate = c.DisableDate;

			return result;
		}
		public EditCustomerViewModel GetEditCustomerViewModel(int id)
		{
			EditCustomerViewModel result = new EditCustomerViewModel();
			Customer editingItem = null;

			editingItem = _db.Customers.Find(id);

			result = Mapper.Map<EditCustomerViewModel>(editingItem);

			return result;
		}


		public ManageCustomerViewModel GetManageCustomerViewModel(int id)
		{
			ManageCustomerViewModel result = new ManageCustomerViewModel();
			Customer dbCustomer = null;
			dbCustomer = _db.Customers.Find(id);

			result.ID = id;
			result.Name = dbCustomer.Name;
			result.IsEnabled = dbCustomer.IsEnabled;

			return result;
		}
	}
}
